modules/ravestate/activation.py

```diff
@@ -1,16 +1,19 @@
 # Ravestate class which encapsualtes the activation of a single state
+from typing import List, Set
 
 from ravestate import state
 from ravestate import wrappers
 from ravestate import icontext
 from threading import Thread
 
+from ravestate.constraint import Signal, Constraint, s
+
 
 class StateActivation:
 
     def __init__(self, st: state.State, ctx: icontext.IContext):
         self.state_to_activate = st
-        self.unfulfilled = [set(trigger_clause) for trigger_clause in st.triggers]
+        self.unfulfilled: Constraint = st.triggers
```

Comment:
Create a deep copy of the constraint here, otherwise `st.triggers` it is single-use 😋 
```suggestion
        self.unfulfilled: Constraint = Disjunct() | st.triggers
```

Reply:
Ah, I'm just realising my suggestion will copy everything except the actual signal objects 🤕